src/query/functions/utils/heap.go

```diff
@@ -33,17 +35,34 @@ type ValueIndexPair struct {
 type lessFn func(ValueIndexPair, ValueIndexPair) bool
 
 func maxHeapLess(i, j ValueIndexPair) bool {
-	if i.Val == j.Val {
+	if i.Val == j.Val || math.IsNaN(i.Val) && math.IsNaN(j.Val) {
```

Comment:
nit: These conditionals are very hard to read and understand, can you make them more clear and verbose?

```suggestion
func maxHeapLess(i, j ValueIndexPair) bool {	func maxHeapLess(i, j ValueIndexPair) bool {
	if math.IsNaN(i.Val) {
		if math.IsNaN(j.Val) {
			return i.Index > j.Index
		}
		
		return true
	}
	
	if math.IsNaN(j.Val) {
		return true
	}
	
	if i.Val == j.Val {
		return i.Index > j.Index
	}
	
	return i.Val < j.Val
}
```

Reply:
I'm not sure that having more code makes it more clear (especially in this case). I believe that logical `and` and `or` operators have their place and were added to programming languages for good reasons. I highly doubt that avoiding them makes code more clear.

Reply:
As a reviewer, it took a while to parse the logic as is; would recommend changing it to be more straightforward

Reply:
Simplified them and added some documentation to make it more clear.